packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -147,11 +191,78 @@ export class RestAPIClient {
 			},
 		};
 
-		const response: MySitesResponse = await this.sendRequest(
-			this.getRequestURL( '1.1', '/me/sites' ),
-			params
-		);
+		return await this.sendRequest( this.getRequestURL( '1.1', '/me' ), params );
+	}
+
+	/**
+	 * Closes the account.
+	 *
+	 * Prior to the account being closed, this method performs
+	 * multiple checks to ensure the operation is safe and that
+	 * the correct account is being closed.
+	 *
+	 * The userID, username and email of the account that is
+	 * authenticated via the bearer token is checked against the
+	 * supplied parameters.
+	 *
+	 * @returns {Promise<boolean>} True if account closure was successful. False otherwise.
+	 */
+	async closeAccount(
+		userID: number,
+		username: string,
+		email: string
```

Comment:
I love the validation we do with this! We are asking consumers to essentially double-check the details of the account they expect to be deleted.

Small, optional, stylistic comment: can we convert all these parameters to properties on a single object parameter, maybe called `expectedAccountDetails`? Right now, looking at just the signature, it's confusing why we need this information when calling (because the bearer token and the class constructor should contain all the account-specific information.) Having it all in one descriptively named object might make it a bit more clear to consumers that they're providing "expected" account information for a double-check.

Reply:
> can we convert all these parameters to properties on a single object parameter,

I agree. Will make the change in the next commit.